libarchive/archive_write_set_format_zip.c

```diff
@@ -520,6 +520,14 @@ archive_write_zip_header(struct archive_write *a, struct archive_entry *entry)
 	struct archive_string_conv *sconv = get_sconv(a, zip);
 	int ret, ret2 = ARCHIVE_OK;
 	mode_t type;
+#if defined(_WIN32)
+	/* On Windows use MS-DOS value as internal Windows zip archiver does
+	 * Fixes charset problems like https://sourceforge.net/p/sevenzip/bugs/2463/ */
+    int create_os = 0;
+#else
+    // Use UNIX value in all other cases
+	int create_os = 3;
+#endif
```

Comment:
It might be worth leaving a reference describing where these constants (0 and 3) come from, eg section "4.4.2 version made by (2 bytes)" in
https://pkwaredownloads.blob.core.windows.net/pkware-general/Documentation/APPNOTE-6.3.9.TXT

Reply:
fixed